pkg/k8s/policy.go

```diff
@@ -95,9 +95,33 @@ func AuthorizationsForResource(ctx context.Context, k8sAPI *KubernetesAPI, names
 		os.Exit(1)
 	}
 
+	allServersInNamespace := map[string]*serverv1beta1.ServerList{}
+
 	for _, p := range policies.Items {
 		target := p.Spec.TargetRef
-		if target.Kind == ServerKind && target.Group == PolicyAPIGroup {
+		if target.Kind == NamespaceKind && target.Group == K8sCoreAPIGroup {
+			serverList, ok := allServersInNamespace[p.Namespace]
+			if !ok {
+				serverList, err = k8sAPI.L5dCrdClient.ServerV1beta1().Servers(p.Namespace).List(ctx, metav1.ListOptions{})
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to get Servers for Namespace/%s: %s\n", p.Namespace, err)
+					continue
+				}
+
+				allServersInNamespace[p.Namespace] = serverList
+			}
+
+			for _, server := range serverList.Items {
+				if serverIncludesPod(server, pods) {
+					results = append(results, Authorization{
```

Comment:
I think we need a `break` after this line in case multiple servers in the namespace include the pods, to prevent the authorization from being included in the results multiple times.

Reply:
Don't think so. We have the similar behaviour [here](https://github.com/linkerd/linkerd2/pull/9337/files#diff-8fe810c43c25c4818b668bf4d2fc343db28faedd84e9304077aa109e12bd52e4L81) at line 81 where several `Server`s can be selected by selector.

By the way, I tested it in our test env:
```
ROUTE   SERVER       AUTHORIZATION_POLICY      SERVER_AUTHORIZATION    
*       website-amb                            linkerd-healthchecks-allow  
*       website-srv                            linkerd-healthchecks-allow  
*       website-amb  linkerd-ingress-allow                             
*       website-srv  linkerd-ingress-allow                             
*       website-amb  linkerd-monitoring-allow                          
*       website-srv  linkerd-monitoring-allow                          
```

As you could see, we have two servers for one POD (deployment with two replicas exactly), and there is no duplication because all information is valuable. ` linkerd-ingress-allow` and `linkerd-monitoring-allow ` policies target the same namespace.

Reply:
If you use break, then we receive the following results:
```
ROUTE   SERVER       AUTHORIZATION_POLICY      SERVER_AUTHORIZATION    
*       website-amb                            linkerd-healthchecks-allow  
*       website-srv                            linkerd-healthchecks-allow  
*       website-amb  linkerd-ingress-allow                             
*       website-amb  linkerd-monitoring-allow     
```

And we lose the information that policies are applied on `website-srv`

Reply:
ah, you are correct.  thanks!